eZ/Publish/Core/REST/Server/Output/ValueObjectVisitor/VersionTranslationInfo.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\REST\Server\Output\ValueObjectVisitor;
+
+use eZ\Publish\Core\REST\Common\Output\ValueObjectVisitor;
+use eZ\Publish\Core\REST\Common\Output\Generator;
+use eZ\Publish\Core\REST\Common\Output\Visitor;
+
+/**
+ * Version value object visitor.
+ */
+class VersionTranslationInfo extends ValueObjectVisitor
+{
+    /**
+     * Visit struct returned by controllers.
+     *
+     * @param \eZ\Publish\Core\REST\Common\Output\Visitor $visitor
+     * @param \eZ\Publish\Core\REST\Common\Output\Generator $generator
+     * @param \eZ\Publish\Core\REST\Server\Values\VersionTranslationInfo $data
+     */
+    public function visit(Visitor $visitor, Generator $generator, $data)
+    {
+        $versionInfo = $data->getVersionInfo();
+        if (empty($versionInfo->languageCodes)) {
+            return;
+        }
+
+        $generator->startObjectElement('VersionTranslationInfo');
+
+        // single language-independent conditions for deleting Translation
+        $canDelete = count($versionInfo->languageCodes) >= 2 && $versionInfo->isPublished();
```

Comment:
I'm not a huge fan of having this calculated in this place. It would make so much more sense if the TranslationInfo object had a `canDelete` property. But I don't know where it could be done given the code flow. So I'm not asking for a change, but if you happen to have an idea... :)